src/hotspot/share/gc/shenandoah/shenandoahHeap.hpp

```diff
@@ -390,6 +393,14 @@ class ShenandoahHeap : public CollectedHeap {
 
 public:
 
+  void finish_fullgc() {
+    Atomic::add(&_completed_fullgc_cycles, (size_t) 1);
```

Comment:
`Atomic::inc`? Also - this will only be called on a safepoint at the end of full gc? Does it need atomic access?

Reply:
Thanks for looking at this.  This is only written from within full gc.  Maybe declaring the variable volatile is sufficient to assure that the "many" mutator threads that access the variable will see any change written by full-gc control thread.  I followed the Atomic pattern out of an abundance of caution.